nucypher/network/nodes.py

```diff
@@ -855,7 +855,10 @@ def from_tls_hosting_power(cls, tls_hosting_power: TLSHostingPower, *args, **kwa
     # Known Nodes
     #
 
-    def seed_node_metadata(self):
+    def seed_node_metadata(self, as_teacher_uri=False):
+        if as_teacher_uri:
```

Comment:
I think I'd prefer if we give these local names and then use them in each logic branch (to prevent refactor divergence later).